packages/demo/src/components/ActionBox/ActionBox.tsx

```diff
@@ -0,0 +1,66 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import { useState, MouseEvent as ReactMouseEvent, useCallback } from "react";
+import { CopyBlock, irBlack } from "react-code-blocks";
+
+type actionFnRequiredParam<T = unknown, U = unknown> = (option: T) => U;
+type actionFnOptionalParam<T = unknown, U = unknown> = (option?: T) => U;
+
+interface IActionBox<T = unknown, U = unknown> {
+  actionTitle: string;
+  actionFn: actionFnOptionalParam<T, U> | actionFnRequiredParam<T, U>;
+  actionOption: T;
+  actionCode: string;
+  actionFnTestId?: string;
+}
+
+export const ActionBox = <T, U>({
+  actionTitle,
+  actionFn,
+  actionOption,
+  actionCode,
+  actionFnTestId = "",
+}: IActionBox<T, U>): JSX.Element => {
+  const [isShowCode, setIsShowCode] = useState(false);
+
+  const onShowCode = useCallback(
+    (event: ReactMouseEvent) => {
+      event.stopPropagation();
+      setIsShowCode((show) => !show);
+    },
+    [setIsShowCode],
+  );
+
+  return (
+    <Box sx={{ display: "flex", flexDirection: "column", alignItems: "center", p: 3, flexGrow: 1 }}>
+      <div>
+        <Button
+          data-testid={actionFnTestId}
+          sx={{ textTransform: "none", mb: 1 }}
+          type="submit"
+          variant="contained"
+          onClick={() => actionFn(actionOption)}
+        >
+          {actionTitle}
+        </Button>
+
+        <Button
+          placeholder="show-code"
+          sx={{ textTransform: "none", mb: 1 }}
+          title="show-code"
+          type="button"
+          variant="text"
+          onClick={onShowCode}
+        >
+          &lt;/&gt;
+        </Button>
+      </div>
+
+      {isShowCode && (
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-expect-error
```

Comment:
Well because of this type error from CodeBlock component: 
```
src/components/ActionBox/ActionBox.tsx:61:20 - error TS2322: Type '{ codeBlock: boolean; showLineNumbers: true; wrapLines: true; language: string; text: string; theme: { lineNumberColor: string; lineNumberBgColor: string; backgroundColor: string; textColor: string; ... 29 more ...; numberColor: string; }; }' is not assignable to type 'IntrinsicAttributes & ExecutionProps & RefAttributes<({ theme, text, codeBlock, customStyle, onCopy, ...rest }: CopyBlockProps) => Element>'.
  Property 'codeBlock' does not exist on type 'IntrinsicAttributes & ExecutionProps & RefAttributes<({ theme, text, codeBlock, customStyle, onCopy, ...rest }: CopyBlockProps) => Element>'.

61         <CopyBlock codeBlock={true} showLineNumbers wrapLines language="typescript" text={code} theme={irBlack} />
                      ~~~~~~~~~


Found 1 error in src/components/ActionBox/ActionBox.tsx:61
```

Reply:
This is strange. Maybe there is some spelling error? I'll take a look later on the code when I test it.

Reply:
Did you test this?

Reply:
There are better options for supporting code render:
https://github.com/FormidableLabs/react-live
https://github.com/react-syntax-highlighter/react-syntax-highlighter

Reply:
![Screenshot_20231026_101450](https://github.com/CryptKeeperZK/crypt-keeper-extension/assets/36774944/efa21b8f-db25-40a0-989b-61c3ec20d927)
I used the first one `react-live`.